inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java

```diff
@@ -278,6 +282,9 @@ public void resiterMeta(JobProfile jobConf) {
             return;
         }
         String[] env = jobConf.get(JOB_FILE_META_ENV_LIST).split(COMMA);
+        if (env.length > 0) {
```

Comment:
Why is metadata required when there are more than one environment?

Reply:
> Why is metadata required when there are more than one environment?

MetaData is used for some special scenes, such as k8s log.

Reply:
If the value of `JOB_FILE_META_ENV_LIST` is "local" or others, but not "k8s", then `needMetaData` will also be true. Is this as expected?

Reply:
> If the value of `JOB_FILE_META_ENV_LIST` is "local" or others, but not "k8s", then `needMetaData` will also be true. Is this as expected?

Okay, you are right